x-pack/plugins/triggers_actions_ui/public/application/sections/rules_list/components/rules_list_notify_badge.tsx

```diff
@@ -79,6 +84,26 @@ const getNextRuleSnoozeSchedule = (rule: { snoozeSchedule?: RuleSnooze }) => {
   return nextSchedule;
 };
 
+// Gets the next occurrence date string relative to the current date
+// Falls back to dtstart in case there is no more next occurrence
+const getNextOccurrenceDateString = (rRule: SnoozeRRule) => {
+  if (moment().isBefore(rRule.dtstart)) {
+    return rRule.dtstart;
+  }
+  try {
+    const recurrenceRule = getRRuleFromSnooze(rRule);
+    const recurrenceDate = recurrenceRule.after(new Date(), true);
+    // If no next occurrence, presumably there are no more occurrences,
+    // therefore, show last occurrence. Falling back to dtstart
+    if (!recurrenceDate) {
+      return recurrenceRule.before(new Date(), true)?.toISOString() || rRule.dtstart;
+    }
```

Comment:
I'm unsure if this is what we want for schedules that have run out of occurrences. But I think displaying the last occurrence makes some sense. 

Unless we have something that cleans up these schedules (I vaguely remember there's a system like this)

Reply:
Yeah we have garbage collection on schedules so this is fine